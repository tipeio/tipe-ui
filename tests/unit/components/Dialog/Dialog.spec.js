import { shallowMount } from '@vue/test-utils'
import tipeTestUtils from '~/tests/TipeTestUtils'
import TipeDialog from '@/components/Dialog'
import faker from 'faker'

describe('Dialog', () => {
  describe('<template>', () => {
    tipeTestUtils.test.snapshot(TipeDialog)

    it('<root> - has correct data-tipe-iu attibute', () => {
      const wrapper = shallowMount(TipeDialog)

      expect(wrapper.attributes()['data-tipe-ui']).toBe('TipeDialog')
    })
  })

  describe(':props', () => {
    it(':title - should render', () => {
      const propsData = { title: faker.lorem.sentence() }
      const wrapper = shallowMount(TipeDialog, { propsData })

      expect(wrapper.find('h1').text()).toEqual(propsData.title)
    })

    it(':message - should render', () => {
      const propsData = { message: faker.lorem.sentence() }
      const wrapper = shallowMount(TipeDialog, { propsData })

      expect(wrapper.find('p').text()).toEqual(propsData.message)
    })
  })
})
