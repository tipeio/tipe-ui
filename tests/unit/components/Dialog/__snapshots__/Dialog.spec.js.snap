// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog <template> matches previous snapshot 1`] = `
<TipeModal-stub height="15rem" width="15rem" data-tipe-ui="TipeDialog">
  <div class="body">
    <h1></h1>
    <p></p>
    <div class="footer">
      <TipeButton-stub color="info" iconBefore="" iconAfter="" name="" size="medium" tabindex="0">
        Cancel
      </TipeButton-stub>
      <TipeButton-stub color="danger" iconBefore="" iconAfter="" name="" size="medium" tabindex="0">
        CONTINUE
      </TipeButton-stub>
    </div>
  </div>
</TipeModal-stub>
`;
